AC_AttitudeControl: improve tailsitter bodyframe roll control

kd0aij opened the pull request:
Splits the roll command input into bodyframe roll and yaw as a function of Euler pitch.
Behaves similar to INPUT_TYPE_MULTICOPTER when hovering and similar to QACRO in cruise attitudes.

kd0aij commented:
Also split tailsitter input_type: BF_ROLL into 2 different types:
TAILSITTER_INPUT_BF_ROLL_M=2 and TAILSITTER_INPUT_BF_ROLL_P=3
The M version is bodyframe roll with the roll stick controlling bodyframe yaw in hover (like TAILSITTER_INPUT_MULTICOPTER) and the P version has the yaw stick controlling bodyframe yaw in hover (like TAILSITTER_INPUT_PLANE)

rmackay9 commented:
Looks like this is only used by quadplanes so I'm not too concerned about what goes on in this function.  One small request is could we add a note in comments for the declaration in AC_AttitudeControl.h to say, "Only used by quadplane"?

kd0aij commented:
@rmackay9 Thanks... added suggested comments for my two new methods
it looks like input_rate_bf_roll_pitch_yaw_2 and input_rate_bf_roll_pitch_yaw_3 are also used only by quadplanes... I don't think input_rate_bf_roll_pitch_yaw_3 was used anywhere until I added the QACRO flight mode

IamPete1 commented:
looks OK to me, not tested tho

kd0aij commented:
tested in SITL/RF8 with Stryker_quad (copter tailsitter) and AddictionX (legacy tailsitter).

kd0aij commented:
@tridge this is now squashed and ready to merge. Thanks for the review and approval.

rmackay9 commented:
As a side note, I'm not a huge fan of the method names, "input_euler_rate_yaw_euler_angle_pitch_bf_roll_m" and 'input_euler_rate_yaw_euler_angle_pitch_bf_roll_p" but I can't immediately think of a better name.

rmackay9 commented:
Merged, thanks!
